src/internal/obj/minio_client.go

```diff
@@ -40,6 +41,89 @@ func newMinioClientV2(endpoint, bucket, id, secret string, secure bool) (*minioC
 	}, nil
 }
 
+func (c *minioClient) Put(ctx context.Context, name string, r io.Reader) (retErr error) {
+	defer func() { retErr = c.transformError(retErr, name) }()
+	wc := newMinioWriter(ctx, c, name)
+	if _, err := io.Copy(wc, r); err != nil {
```

Comment:
We should probably close here similarly to the microsoft client, then add a TODO to handle the context appropriately.

Reply:
Actually, I think we discussed removing the minio client in 2.0 and just having the amazon client for s3 compatible object storage providers. We might want to figure out if we still want to do that.

Reply:
https://github.com/pachyderm/pachyderm/issues/5919